third_party/terraform/utils/transport.go

```diff
@@ -35,39 +37,56 @@ func sendRequest(config *Config, method, rawurl string, body map[string]interfac
 	reqHeaders.Set("User-Agent", config.userAgent)
 	reqHeaders.Set("Content-Type", "application/json")
 
-	var buf bytes.Buffer
-	if body != nil {
-		err := json.NewEncoder(&buf).Encode(body)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	u, err := addQueryParams(rawurl, map[string]string{"alt": "json"})
+	var res *http.Response
+	err := retryTimeDuration(
+		func() error {
+			var buf bytes.Buffer
+			if body != nil {
+				err := json.NewEncoder(&buf).Encode(body)
+				if err != nil {
+					return err
+				}
+			}
+
+			u, err := addQueryParams(rawurl, map[string]string{"alt": "json"})
+			if err != nil {
+				return err
+			}
+			req, err := http.NewRequest(method, u, &buf)
+			if err != nil {
+				return err
+			}
+
+			req.Header = reqHeaders
+			res, err = config.client.Do(req)
+			if err != nil {
+				return err
+			}
+
+			if err := googleapi.CheckResponse(res); err != nil {
+				googleapi.CloseBody(res)
+				return err
+			}
+
+			return nil
+		},
+		time.Duration(5)*time.Minute,
```

Comment:
why 5 minutes and not the time that the user specified in their timeouts block?

Reply:
I am going to make the changes to route the timeout blocks in https://github.com/GoogleCloudPlatform/magic-modules/pull/986 because they might be a bit expansive. I'll add a todo comment here so that's more obvious.

Reply:
If they have gotten to this point and the function should have either returned an error or a non-nil response object. If both are nil then it's an unexpected situation. I could add to the error message something like
> please file a bug at https://github.com/terraform-providers/terraform-provider-google/issues